nodeadm/api/v1alpha1/nodeconfig_types.go

```diff
@@ -43,10 +44,10 @@ type ClusterDetails struct {
 }
 
 type KubeletOptions struct {
-	// Config is a raw document of a kubelet config that can be provided
-	// by the user to override default generated configurations
+	// Config is a kubelet config that can be provided by the user to override
+	// default generated configurations
 	// https://kubernetes.io/docs/reference/config-api/kubelet-config.v1/
-	Config string `json:"config,omitempty"`
+	Config map[string]runtime.RawExtension `json:"config,omitempty"`
```

Comment:
`map[string]interface{}` isn't allowed by conversion-gen, so this is on of the next best suggestions to achieve the same end usage. the internal usages are a bit more complex based on instances where u cannot directly merge `map[string]interface{}` with `map[string]runtime.RawExtension`

Reply:
This looks like the way to go. If merging this type with mergo is cumbersome, can we write a custom handler or something? Marshalling to-and-from JSON will work for now but would be nice to clean that up

Reply:
sounds good 👍, as long as the use case works we can hash this out